torch/csrc/dynamo/eval_frame.c

```diff
@@ -337,6 +367,44 @@ static struct PyGetSetDef CacheEntry_properties[] = {
     {NULL}};
 
 
+static CacheEntry* extract_cache_entry(ExtraState* extra_state);
+
+static PyObject* CacheEntry_invalidate(CacheEntry* self, PyObject* args) {
+  // handle case where CacheEntry is the first entry in self->extra_state
+  CacheEntry* first_cache_entry = extract_cache_entry(self->extra_state);
+  if (first_cache_entry != NULL && first_cache_entry == self) {
+    DEBUG_CHECK((PyObject*)self->prev == Py_None);
+    // will be decremented later on, so increment now
+    Py_INCREF(self->next);
+    self->extra_state->cache_entry = self->next;
+  }
+
+  // delete this CacheEntry from the linked list
+  if ((PyObject*)self->next != Py_None) {
+    self->next->prev = self->prev;
+  }
+  if ((PyObject*)self->prev != Py_None) {
+    // steal reference
+    self->prev->next = self->next;
+    self->prev = (CacheEntry*)Py_None;
+  } else {
+    Py_DECREF(self->next);
+  }
+  Py_INCREF(Py_None);
```

Comment:
Is `Py_RETURN_NONE` - https://docs.python.org/3/c-api/none.html#c.Py_RETURN_NONE better?